backends/gdx-backends-gwt/src/com/badlogic/gwtref/gen/ReflectionCacheGenerator.java

```diff
@@ -16,31 +16,207 @@
 
 package com.badlogic.gwtref.gen;
 
+import com.google.gwt.core.ext.BadPropertyValueException;
+import com.google.gwt.core.ext.ConfigurationProperty;
 import com.google.gwt.core.ext.Generator;
 import com.google.gwt.core.ext.GeneratorContext;
 import com.google.gwt.core.ext.TreeLogger;
 import com.google.gwt.core.ext.UnableToCompleteException;
+import com.google.gwt.core.ext.typeinfo.JArrayType;
 import com.google.gwt.core.ext.typeinfo.JClassType;
+import com.google.gwt.core.ext.typeinfo.JField;
+import com.google.gwt.core.ext.typeinfo.JMethod;
+import com.google.gwt.core.ext.typeinfo.JPackage;
+import com.google.gwt.core.ext.typeinfo.JPrimitiveType;
+import com.google.gwt.core.ext.typeinfo.JType;
 import com.google.gwt.core.ext.typeinfo.TypeOracle;
-import static com.google.gwt.core.ext.TreeLogger.*;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import static com.google.gwt.core.ext.TreeLogger.ERROR;
+import static com.google.gwt.core.ext.TreeLogger.Type;
 
 public class ReflectionCacheGenerator extends Generator {
+
+	private GeneratorContext context;
+	private TreeLogger logger;
+
 	@Override
 	public String generate (TreeLogger logger, GeneratorContext context, String typeName) throws UnableToCompleteException {
+		this.context = context;
+		this.logger = logger;
 		TypeOracle oracle = context.getTypeOracle();
 		assert (oracle != null);
-		JClassType type = oracle.findType(typeName);
-		if (type == null) {
+		JClassType jType = oracle.findType(typeName);
+		if (jType == null) {
 			logger.log(ERROR, "Couldn't find type '" + typeName + "'");
 			throw new UnableToCompleteException();
 		}
 
-		if (type.isInterface() == null) {
+		if (jType.isInterface() == null) {
 			logger.log(ERROR, "Type '" + typeName + "' must be an interface");
 			throw new UnableToCompleteException();
 		}
 
-		ReflectionCacheSourceCreator source = new ReflectionCacheSourceCreator(logger, context, type);
+		TypeOracle typeOracle = context.getTypeOracle();

```

Comment:
level of abstraction, would be easier to grok if type collection had it's own class or encapsulated in method.